x-pack/legacy/plugins/uptime/public/state/api/snapshot.ts

```diff
@@ -18,27 +19,22 @@ interface ApiRequest {
 }
 
 export const fetchSnapshotCount = async ({
-  basePath,
   dateRangeStart,
   dateRangeEnd,
   filters,
   statusFilter,
 }: ApiRequest): Promise<Snapshot> => {
-  const url = getApiPath(`/api/uptime/snapshot/count`, basePath);
-  const params = {
+  const queryParams = {
     dateRangeStart,
     dateRangeEnd,
     ...(filters && { filters }),
     ...(statusFilter && { statusFilter }),
   };
-  const urlParams = new URLSearchParams(params).toString();
-  const response = await fetch(`${url}?${urlParams}`);
-  if (!response.ok) {
-    throw new Error(response.statusText);
-  }
-  const responseData = await response.json();
+
+  const responseData = await apiService.get(API_URLS.SNAPSHOT_COUNT, queryParams);
+
   const decoded = SnapshotType.decode(responseData);
-  ThrowReporter.report(decoded);
+  PathReporter.report(decoded);
```

Comment:
Same comment here, let's stick with the `ThrowReporter`. Let's not mix too many things into one PR unless there's an issue with separating them out.

Reply:
we don't need to throw it, because we are not catching it. We are only checking for errors now.